### ISSUE TITLE
`\cs_set:cpx` misinterpreted as `\cs_set:Npx` by index

### ISSUE THREAD

**Issue by niruvt** [latex3/latex3#1636]
I was clearing off unneeded commands from my index. Despite having no occurrences of `\cs_set:Npx` at all, it managed to linger on my index. Adding it to `\DoNotIndex` (of course) had no effect. I checked the line number it pointed to and to my shock that line had `\cs_set:cpx`. Adding `\cs_set:cpx` works for my purpose, but this needs to be fixed.

```
% \iffalse
%<*driver>
\documentclass{l3doc}

\begin{document}
\DocInput{\jobname.dtx}
\end{document}
%</driver>
\fi
% \DoNotIndex{\ExplSyntaxOn,\ExplSyntaxOff,\cs_set:Npx}
%    \begin{macrocode}
%<*pkg>
\ExplSyntaxOn
\cs_set:cpx { foo } { \today }
\ExplSyntaxOff
%</pkg>
%    \end{macrocode}
% \PrintIndex
```

**Comment by josephwright** [comment]
This is a feature of `l3doc`, not of `l3build`: variants are indexed under their parent.

**Comment by niruvt** [comment]
Sorry for reporting at the wrong place.

> variants are indexed under their parent.

Got it, any way to turn it off? If I want to list variants separately, is there a way?

**Comment by niruvt** [comment]
Since we are at it, is there any counterpart of `\DoNotIndex`? Something like `\OnlyIndex`. It is quite a task to manually list down the commands we don't want.

**Comment by niruvt** [comment]
Also, if variants are listed under their parent, then why are they indexed when their parent is removed from the list with `\DoNotIndex`? It doesn't obey the pattern.

**Comment by josephwright** [comment]
> Got it, any way to turn it off? If I want to list variants separately, is there a way?

I don't think so - it's by-design and `l3doc` is a load of hacks to let us do kernel work, so controls are limited.

**Comment by josephwright** [comment]
> Since we are at it, is there any counterpart of `\DoNotIndex`? Something like `\OnlyIndex`. It is quite a task to manually list down the commands we don't want.

Just list everything :)

**Comment by msyam** [comment]
> > Since we are at it, is there any counterpart of `\DoNotIndex`? Something like `\OnlyIndex`. It is quite a task to manually list down the commands we don't want.
> 
> Just list everything :)

It is indeed rather cumbersome to list every single command used.
Support for regular expressions, prefix matching, or module‑based filtering would be a perfectly reasonable requirement.
Alternatively, it would be helpful if the official source provided a manifest that we could reference directly.
